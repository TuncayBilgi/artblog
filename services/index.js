import { request, gql } from 'graphql-request'

const graphqlAPI= process.env.NEXT_PUBLIC_GRAPHCMS_ENDPOINT

export const getPosts = async () => { 
    const query = gql `
    query MyQuery {
        postsConnection {
            edges {
                node {
                    author {
                        bio
                        name
                        id
                        photo {
                            url
                        }
                    }
              createdAt
              excerpt
              slug
              title
              featureImage {
                url
              }
              categories {
                name
                slug
              }
            }
          }
        }
    }
        ` 

        const result = await request(graphqlAPI, query);
        return result.postsConnection.edges
      };

      export const getRecentPosts = async() => {
        const query = gql`
        query getPostsDetails {
          posts(last: 3, orderBy: createdAt_ASC) {
            createdAt
            title
            slug
            featureImage {
              url
            }
          }
        }
        
        `
        const result = await request(graphqlAPI, query)
        return result.posts

      }

      export const getSimilarPosts = async(categories,slug) => {
        const query = gql`
        query getPostDetails($slug: String! ,$categories : [String!]){
          posts(
            where: { slug_not: $slug, AND: {categories_some: {slug_in: $categories}}}
            last: 3
          ){
            title
            featureImage {
              url
            }
            createdAt
            slug
          }
        }
        `
        const result = await request(graphqlAPI, query, {categories,slug})
        return result.posts

      }

      export const getCategories = async() => {
        const query = gql`
        query GetCategories {
          categories  {
            name
            slug
          }
        }

        `

        const result = await request(graphqlAPI, query)
        return result.categories
      }

      export const getPostDetails = async (slug) => {
        const query = gql`
          query GetPostDetails($slug : String!) {
            post(where: {slug: $slug}) {
              title
              excerpt
              featureImage {
                url
              }
              author{
                name
                bio
                photo {
                  url
                }
              }
              createdAt
              slug
              content {
                raw
              }
              categories {
                name
                slug
              }
            }
          }
        `;
      
        const result = await request(graphqlAPI, query, { slug });
      
        return result.post;
      };

      export const getPostsByCat = async (slug_contains) => {
        const query = gql`
        query MyQuery($slug_contains: String!) {
          postsConnection(where: {categories_some: {slug_contains: $slug_contains}}) {
            edges {
              node {
                author {
                  bio
                  name
                  id
                  photo {
                    url
                  }
                }
                createdAt
                excerpt
                slug
                title
                featureImage {
                  url
                }
                categories {
                  name
                  slug
                }
              }
            }
          }
        }
        `
        ;        
      
        const result = await request(graphqlAPI, query, { slug_contains });
      
        return result.postsConnection.edges;
      };

      export const submitComment = async (obj) => {
        const result = await fetch('/api/comments', {
           method: 'POST',
           headers: { 'Content-Type': 'application/json'},
           body: JSON.stringify(obj)
           });
        console.log('LE RESULT DU BODY', result);
        return result.json();
      }

      export const getComments = async(slug) => {
        const query = gql`
        query GetComments($slug: String!) {
          comments(where :  {post: {slug : $slug}}){
            name
            createdAt
            comment

          }
        }
        `
        const result = await request(graphqlAPI, query,{slug})
        return result.comments
      }


      