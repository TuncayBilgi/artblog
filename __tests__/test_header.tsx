import React from 'react';
//import { act, render, waitFor } from '@testing-library/react';
import renderer,{act} from 'react-test-renderer';
import Header from '../components/Header';
import { getCategories } from '../services';

jest.mock('../services', () => ({
  getCategories: jest.fn(),
}));



describe('Header', () => {

  beforeEach(() => {
    (getCategories as jest.Mock).mockResolvedValue([{ name: 'category', slug: 'slug' }, { name: 'category3', slug: 'slug2' }]);
  });

  it('renders correctly', async () => {
      const element = renderer.create(<Header/>).toJSON();
      expect(element).toMatchSnapshot();
  });
});