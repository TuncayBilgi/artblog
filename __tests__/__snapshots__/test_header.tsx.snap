// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="container mx-auto px-10 mb-4"
  id="header"
>
  <div
    className=" border-black w-full inline-block py-6 border-b-2"
  >
    <div
      className="md:float-left block"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="cursor-pointer font-bold text-4xl text-white inline-block"
        >
          ArtBlog
        </span>
      </a>
    </div>
    <div
      className="hidden md:float-left md:contents"
    />
  </div>
</div>
`;
